ShadowDOM/explainer.md

```diff
@@ -371,8 +448,37 @@ A challenge that arises is dealing with scopes and idrefs. If a declarative styl
 The script version of this already exists via the [adoptedStylesheets](https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot/adoptedStyleSheets) property: 
 ```html
 import sheet from './styles.css' assert { type: 'css' }; // or new CSSStyleSheet(); 
-shadowRoot.adoptedStyleSheets = [sheet];
+shadowRoot.adoptedStyleSheets = [sheet]; 
+```
+
+## Polyfills
+
+Web developers often seek polyfills to allow them to use new web platform features while falling back gracefully in user agents where such features are not supported. A common strategy is to use JavaScript for polyfills. An example of this could be the following: 
+
+```html
+<script> 
+  function supportsDeclarativeAdoptedStyleSheets() { 
+    return document.createElement('template')["adoptedStyleSheets"] == undefined; 
+  } 
+
+  if (!supportsDeclarativeAdoptedStyleSheets()) { 
+    // AdoptedStyleSheets is not supported on <template> - apply polyfill. This polyfill could be an injected <link> tag. 
+  } 
+</script> 
 ```
+
+For the Declarative CSS Modules approach, [one suggestion](https://github.com/whatwg/html/issues/10673#issuecomment-2453512552) is to bind the `<link>` tag's `href` attribute value to the CSS module identifier and add a new attribute (`noadoptedstylesheets`) to avoid double-applying stylesheets. 
```

Comment:
```suggestion
There was also a [suggestion](https://github.com/whatwg/html/issues/10673#issuecomment-2453512552) for adding browser support to enable falling back to a normal `<link>` tag without the use of script, by binding the `<link>` tag's `href` attribute value to the CSS module identifier and adding a new attribute (`noadoptedstylesheets`) to avoid double-applying stylesheets. 
```

Optional suggestion for clarifying the purpose of this alternative.